docs/Development_Documentation/04_Assets/03_Working_with_Thumbnails/01_Image_Thumbnails.md

```diff
@@ -47,6 +46,26 @@ This path can then be directly used to display the image in a `<img />` or `<pic
 <?php } ?>
 ```
 
+**An Alternative Adapter**
+Since build **3999** you can use the new adapter: `pimcore/lib/Pimcore/Image/Adapter/ImageMagick.php` instead of the standard `pimcore/lib/Pimcore/Image/Adapter/Imagick.php`. 
+The main difference: `ImageMagick` is using `convert` and `composite` CLI tools to manage thumbnails (it gives a better control of the generation process).
+
+To try the new adapter add a new [dependency injection](../../10_Extending_Pimcore/03_Dependency_Injection.md) declaration in the `website/config/di.php` file. 
+
+If `convert` and `composite` are accessible globally:
+
+```php
+\Pimcore\Image\Adapter::class => DI\object('Pimcore\Image\Adapter\ImageMagick')
+```
+
+You can also specify the path to tools, like below:
+```php
+\Pimcore\Image\Adapter::class => DI\object('Pimcore\Image\Adapter\ImageMagick')
+        ->method('setConvertScriptPath', '/usr/bin/convert')
```

Comment:
I think this configurations should be optional. 
Pimcore has an locator for binaries, which should be used in `getConvertScriptPath()` and `getCompositeCommand()` if the path was not set manually before. 

This is how to use the locator: 
```php
$path = \Pimcore\Tool\Console::getExecutable("convert");
$path = \Pimcore\Tool\Console::getExecutable("composite");
```